libirods_rule_engine_plugin-indexing.cpp

```diff
@@ -217,9 +219,9 @@ namespace {
                             // verify that this is not new metadata with a query and set a flag
                             collection_metadata_is_new = !idx.metadata_exists_on_collection(
                                                              object_path,
-                                                             avu_inp->arg3,
-                                                             avu_inp->arg4,
-                                                             avu_inp->arg5);
+                                                             NULL_PTR_GUARD(avu_inp->arg3),
+                                                             NULL_PTR_GUARD(avu_inp->arg4),
```

Comment:
the `NULL_PTR_GUARD` seems like it should only apply to the units value since it is optional.

isn't it an error to pass null or empty strings to the attribute name and value? I feel the name and value should be checked and on a bad value, return, log, or throw an error.

thoughts?